Mathlib/Order/BourbakiWitt.lean

```diff
@@ -0,0 +1,227 @@
+  · constructor
+    · exact ⟨bot_isAdmissible.base_isLeast.1, Or.inl (bot_isAdmissible.base_isLeast.2 hy.mem_bot)⟩
+    · exact fun y h ↦ bot_isAdmissible.base_isLeast.2 h.1
+  · rintro _ ⟨z, ⟨hz, (hzy | hyz)⟩, rfl⟩ <;>
+      refine ⟨map_mem_bot hz, ?_⟩
+    · rcases le_iff_lt_or_eq.1 hzy with (hzy | rfl)
+      · left; exact hy.map_le hz hzy
+      · right; exact le_refl _
+    · right; exact le_trans hyz (f.le_map z)
+  · intro c hc
+    refine ⟨bot_isAdmissible.cSup_mem _ (subset_trans hc (sep_subset _ _)), ?_⟩
+    · by_cases h : ∀ z ∈ c, z ≤ y
+      · left; apply cSup_le c y h
+      · push_neg at h
+        rcases h with ⟨z, hz, hzy⟩
+        obtain h' := Or.resolve_left (hc hz).2 hzy
+        right
+        apply le_trans h' (le_cSup _ _ hz)
+
+lemma setOf_isExtremePt_eq_bot : {y | IsExtremePt x f y} = bot x f := by
+  apply sub_bot_eq_bot _ (fun _ h ↦ h.mem_bot)
+  apply IsAdmissible.mk
+  · constructor
+    · refine ⟨bot_isAdmissible.base_isLeast.1, ?_⟩
+      intro y hy hyx
+      exfalso
+      exact lt_irrefl x (lt_of_le_of_lt (bot_isAdmissible.base_isLeast.2 hy) hyx)
+    · exact fun y h ↦ bot_isAdmissible.base_isLeast.2 h.1
+  · rintro _ ⟨y, hy, rfl⟩
+    refine ⟨map_mem_bot hy.mem_bot, ?_⟩
+    intro z hz hzy
+    have hz' := hz
+    rw [← eq_bot hy] at hz'
+    rcases hz' with ⟨_, (hz' | hz')⟩
+    · rcases le_iff_lt_or_eq.1 hz' with (hz' | rfl)
+      · exact le_trans (hy.map_le hz hz') (f.le_map y)
+      · exact le_refl (f z)
+    · exfalso
+      exact lt_irrefl z (lt_of_lt_of_le hzy hz')
+  · intro c hc
+    refine ⟨bot_isAdmissible.cSup_mem _ (subset_trans hc (fun _ h ↦ h.mem_bot)), ?_⟩
+    intro y hy hy'
+    by_cases h : ∀ z ∈ c, f z ≤ y
+    · exfalso
+      apply lt_irrefl y (lt_of_lt_of_le hy' ?_)
+      apply cSup_le
+      intro z hz
+      apply le_trans (f.le_map z) (h z hz)
+    · push_neg at h
+      obtain ⟨z, hz, hzy⟩ := h
+      have h := hy
+      rw [← eq_bot (hc hz)] at h
+      obtain (hyz | rfl) := le_iff_lt_or_eq.1 (Or.resolve_right h.2 hzy)
+      · exact le_trans ((hc hz).map_le hy hyz) (le_cSup _ _ hz)
+      · have hc' := bot_isAdmissible.cSup_mem _ (subset_trans hc ((fun _ h ↦ h.mem_bot)))
+        rw [← eq_bot (hc hz)] at hc'
+        obtain ⟨_, (hc' | hc')⟩ := hc'
+        · exfalso
+          apply lt_irrefl y (lt_of_lt_of_le hy' hc')
+        · exact hc'
+
+lemma mem_bot_iff_isExtremePt {y : α} : y ∈ bot x f ↔ IsExtremePt x f y := by
+  rw [← setOf_isExtremePt_eq_bot]
+  rfl
+
+lemma bot_isChain : IsChain (· ≤ ·) (bot x f) := by
+  intro y hy z hz _
+  rw [mem_bot_iff_isExtremePt] at hy
+  rw [← eq_bot hy] at hz
+  obtain ⟨_, (hz | hz)⟩ := hz
+  · right; exact hz
+  · left; exact le_trans (f.le_map y) hz
+
+end BourbakiWitt
+
+end
+
+open BourbakiWitt Function
+
+/- **The Bourbaki-Witt Theorem**: If `α` is a chain complete partial order and `f : α → α` is
+inflationary, then `f` has a fixed point -/
+theorem bourbaki_witt {α : Type} [Nonempty α] [ChainCompletePartialOrder α]
```

Comment:
We always use a "boring" type-based name for theorems, which would look something like:
```suggestion
theorem nonempty_fixedPoints_of_inflationary {α : Type} [Nonempty α] [ChainCompletePartialOrder α]
```

Reply:
One option is to put this into the `ChainCompletePartialOrder` namespace, where we can already have `variable {α : Type*} [ChainCompletePartialOrder α]`.